spec/models/manageiq/providers/kubernetes/container_manager_spec.rb

```diff
@@ -47,4 +47,40 @@
       expect(job.zone).to eq("default")
     end
   end
+
+  context "Proxy Configuration" do
```

Comment:
@simon3z I wouldn't bother going through the full stack testing you're doing here. This is all the responsibility of the respective middleware clients and doesn't need to create a proxy just to test that the correct option is passed.

I suggest you simply `expect(VMDB::Util).to receive(:http_proxy_uri).and_return(...)` (expect, because you want to ensure that http_proxy_uri is used) and `expect(Kubeclient::Client).to receive(:new).with(<arguments you expect, especially the http_proxy_uri>)` I'm assuming the client does all the work you recreated below on initialization?